baselines/pFedHN/pFedHN/client.py

```diff
@@ -0,0 +1,134 @@
+"""Client Handling."""
+
+from collections import OrderedDict
+
+import flwr as fl
+import torch
+
+from pFedHN.models import CNNTarget
+from pFedHN.trainer import train
+
+
+# pylint: disable=too-many-instance-attributes
+class FlowerClient(fl.client.NumPyClient):
+    """Flower client for federated learning.
+
+    Args:
+        cid (str): The client ID.
+        trainloader (torch.utils.data.DataLoader): DataLoader for training data.
+        testloader (torch.utils.data.DataLoader): DataLoader for test data.
+        cfg (Config): Hydra Configuration.
+
+    Attributes
+    ----------
+        cid (str): The client ID.
+        trainloader (torch.utils.data.DataLoader): DataLoader for training data.
+        testloader (torch.utils.data.DataLoader): DataLoader for test data.
+        device (torch.device): The device to run the model on.
+        epochs (int): Number of training epochs.
+        n_kernels (int): Number of convolutional kernels.
+        learning_rate (float): Learning rate.
+        weight_decay (float): Weight decay.
+        net (CNNTarget): Target neural network model.
+    """
+
+    def __init__(self, cid, trainloader, testloader, cfg) -> None:
+        super().__init__()
+
+        self.cid = cid
+        self.trainloader = trainloader
+        self.testloader = testloader
+        self.device = torch.device(cfg.model.device)
```

Comment:
Let's not do that like this. Client's should automatically figure out which device to use based on the `client_resources` specified when calling `start_simulation`. With a line like the one below, your clients can easily switch between CPU,GPU based on what the simulation is doing:
```
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```

Reply:
You can use that in `train()` also.